src/shared_modules/utils/rocksDBWrapper.hpp

```diff
@@ -0,0 +1,158 @@
+            }
+
+            const auto status {m_db->Put(rocksdb::WriteOptions(), key, value)};
+            if (!status.ok())
+            {
+                throw std::runtime_error("Error putting data: " + status.ToString());
+            }
+        }
+
+        /**
+         * @brief Get a value from the database.
+         *
+         * @param key Key to get.
+         * @param value Value to get
+         */
+        void get(const std::string& key, std::string& value)
+        {
+            const auto status {m_db->Get(rocksdb::ReadOptions(), key, &value)};
+            if (status.IsNotFound())
+            {
+                throw std::invalid_argument("Key not found: " + key);
+            }
+            else if (!status.ok())
+            {
+                throw std::runtime_error("Error getting data: " + status.ToString());
+            }
+        }
+
+        /**
+         * @brief Delete a key-value pair from the database.
+         *
+         * @param key Key to delete.
+         */
+        void delete_(const std::string& key) // NOLINT
+        {
+            const auto status {m_db->Delete(rocksdb::WriteOptions(), key)};
+            if (!status.ok())
+            {
+                throw std::runtime_error("Error deleting data: " + status.ToString());
+            }
+        }
+
+        /**
+         * @brief Get the last key-value pair from the database.
+         *
+         * @return std::pair<std::string, std::string> Last key-value pair.
+         */
+        std::pair<std::string, std::string> getLastKeyValue()
+        {
+            std::string lastKey {};
+            std::string lastValue {};
+            std::unique_ptr<rocksdb::Iterator> it(m_db->NewIterator(rocksdb::ReadOptions()));
+            for (it->SeekToLast(); it->Valid(); it->Prev())
+            {
+                lastKey = it->key().ToString();
+                lastValue = it->value().ToString();
+                break;
+            }
+            return {lastKey, lastValue};
+        }
+
+        /**
+         * @brief Delete all key-value pairs from the database.
+         */
+        void deleteAll()
+        {
+            rocksdb::WriteBatch batch;
+            std::unique_ptr<rocksdb::Iterator> it(m_db->NewIterator(rocksdb::ReadOptions()));
+            for (it->SeekToFirst(); it->Valid(); it->Next())
+            {
+                batch.Delete(it->key());
+            }
+            const auto status {m_db->Write(rocksdb::WriteOptions(), &batch)};
+            if (!status.ok())
+            {
+                throw std::runtime_error("Error deleting data: " + status.ToString());
+            }
+        }
+
+        /**
+         * @brief Get an iterator pointing to the key-value pair with the given key.
+         *
+         * @return std::unique_ptr<rocksdb::Iterator> Iterator pointing to the key-value pair with the given key.
+         */
+        std::unique_ptr<rocksdb::Iterator> seek(const std::string& key)
```

Comment:
check if the key is empty and throw.

Reply:
An iterator pattern could be a great solution for this case.

Ref:
https://github.com/wazuh/wazuh/blob/17469f88206a90e54e0d5f6ee81a0a04996ada1f/src/data_provider/src/packages/rpmPackageManager.cpp#L84